public/js/components/SourceTabs.js

```diff
@@ -125,12 +125,13 @@ const SourceTabs = React.createClass({
       click: () => tabs.forEach(closeTab)
     };
 
-    showMenu(e, [
-      closeTabMenuItem,
-      closeOtherTabsMenuItem,
-      closeTabsToRightMenuItem,
-      closeAllTabsMenuItem
-    ]);
+    showMenu(e, buildMenu([
+      { item: closeTabMenuItem },
+      { item: closeOtherTabsMenuItem, hidden: () => tabs.size === 1 },
+      { item: closeTabsToRightMenuItem, hidden: () =>
+         tabs.some((t, i) => t === tab && (tabs.size - 1) === i) },
```

Comment:
so for every loop over the tabs items, i'm checking two things

1. `t === tab`. If the current iteration tab is the selected one.

2.  `(tabs.size - 1) === i)` . if the tab item is the last one in the array 

 If these 2 conditions are satisfied , a `true` is returned to `some` meaning that this item does not have tab items to its right.